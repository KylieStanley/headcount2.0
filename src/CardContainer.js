import React from 'react';
import Card from './Card';
import './CardContainer.css'

const CardContainer = (props) => {
  
    return (
      <div className="card-container">
        {
          props.data.map((district, i) => {
            return <Card district={district} key={i} />
          })
        }
      </div>
    );
  }

export default CardContainer;
