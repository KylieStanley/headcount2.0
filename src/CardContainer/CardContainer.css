.card-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.card {
  align-items: center;
  border-radius: 20px;
  border: 1px solid black;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 5px;
  padding: 8px;
  text-align: center;
  width: 250px;
}