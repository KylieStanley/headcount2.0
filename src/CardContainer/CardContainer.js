import React from 'react';
import Card from '../Card/Card';
import './CardContainer.css'

const CardContainer = ({ data, selectCard, selectedCards }) => {

  const allCards = data.map((district, i) => {
                  return <Card district={district} key={i} selectCard={selectCard} selectedCards={selectedCards}/>
                })

  return (
    <div className="card-container">
        { allCards }
    </div>
  );
}

export default CardContainer;
