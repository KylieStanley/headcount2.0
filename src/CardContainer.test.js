import React from 'react';
import Data from './data/kindergartners_in_full_day_program.js';
import CardContainer from './CardContainer.js';
import {shallow} from 'enzyme';

describe('CardContainer', () => {
  it('should match the snapshot with data passed in correctly', () => {
    const wrapper = shallow( <CardContainer 
      data={[{location: "COLORADO",
              stats: {2004: .5}
            }]}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});