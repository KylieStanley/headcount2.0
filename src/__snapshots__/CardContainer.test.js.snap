// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot with data passed in correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    data={
        Array [
            Object {
              "location": "COLORADO",
              "stats": Object {
                "2004": 0.5,
              },
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          district={
                    Object {
                              "location": "COLORADO",
                              "stats": Object {
                                "2004": 0.5,
                              },
                            }
          }
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "district": Object {
            "location": "COLORADO",
            "stats": Object {
              "2004": 0.5,
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            district={
                        Object {
                                    "location": "COLORADO",
                                    "stats": Object {
                                      "2004": 0.5,
                                    },
                                  }
            }
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "district": Object {
              "location": "COLORADO",
              "stats": Object {
                "2004": 0.5,
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
